code/modules/reagents/random_reagent_effect.dm

```diff
@@ -0,0 +1,276 @@
+
+/datum/random_reagent_effect/kill/on_human_life(mob/living/carbon/human/H)
+	H.visible_message("<span class='warning'>\The [H] suddenly collapses!</span>")
+	H.death()
+
+/// --------- Catbeast transformation
+/datum/random_reagent_effect/tf_catbeast
+	name = "TF Catbeast"
+	activation_chance = 1
+	investigative_log = "transform user into catbeast -- "
+
+/datum/random_reagent_effect/tf_catbeast/on_human_life(mob/living/carbon/human/H)
+	if (!iscatbeast(H))
+		H.set_species("Tajaran", transfer_damage = TRUE)
+		H.regenerate_icons()
+		H.emote("me", MESSAGE_HEAR, pick("meows", "mews"))
+		playsound(H, 'sound/voice/catmeow.ogg', 100)
+		log_effect(H, "was transformed into a catbeast")
+
+
+/// --------- Simple mob transformation
+/datum/random_reagent_effect/tf_simplemob
+	name = "TF Simple mob"
+	activation_chance = 4
+	var/picked_mob_type
+
+/datum/random_reagent_effect/tf_simplemob/on_pick()
+	// This is a BIT ugly but eeeh
+	var/generator/rand = generator("num", 0, 150)
+	var/picked_number = rand.Rand()
+	switch(picked_number)
+		if (0 to 100)
+			picked_mob_type = pick(
+				/mob/living/simple_animal/capybara, /mob/living/simple_animal/cat,
+				/mob/living/simple_animal/cat/kitten, /mob/living/simple_animal/cat/snek,
+				/mob/living/simple_animal/chick, /mob/living/simple_animal/chicken,
+				/mob/living/simple_animal/corgi, /mob/living/simple_animal/corgi/puppy,
+				/mob/living/simple_animal/corgi/saint, /mob/living/simple_animal/corgi/sasha,
+				/mob/living/simple_animal/cow, /mob/living/simple_animal/crab,
+				/mob/living/simple_animal/hamster, /mob/living/simple_animal/hostile/retaliate/goat,
+				/mob/living/simple_animal/hostile/retaliate/goat/wooly, /mob/living/simple_animal/parrot,
+				/mob/living/simple_animal/penguin, /mob/living/simple_animal/penguin/chick,
+				/mob/living/simple_animal/rabbit, /mob/living/simple_animal/rabbit/bunny,
+			)
+		if (100 to 130) // Uncommon
+			picked_mob_type = pick(
+				/mob/living/simple_animal/borer, /mob/living/simple_animal/puddi/happy,
+				/mob/living/simple_animal/puddi/anger, /mob/living/simple_animal/spiderbot,
+			)
+		if (130 to 145) // Dangerous
+			picked_mob_type = pick(
+				/mob/living/simple_animal/amogusflash,
+				/mob/living/simple_animal/hostile/asteroid/basilisk, /mob/living/simple_animal/hostile/asteroid/goldgrub,
+				/mob/living/simple_animal/hostile/asteroid/goliath, /mob/living/simple_animal/hostile/asteroid/rockernaut,
+				/mob/living/simple_animal/hostile/bear, /mob/living/simple_animal/hostile/carp,
+				/mob/living/simple_animal/hostile/giant_spider/hunter, /mob/living/simple_animal/hostile/pitbull,
+				/mob/living/simple_animal/slime, /mob/living/simple_animal/slime/adult,
+			)
+		if (145 to 150) // Owned
+			picked_mob_type = pick(
+				/mob/living/simple_animal/hostile/retaliate/clown, /mob/living/simple_animal/hostile/retaliate/cluwne,
+				/mob/living/simple_animal/hostile/retaliate/faguette, /mob/living/simple_animal/hostile/retaliate/mime,
+			)
```

Comment:
Best to avoid overlapping switch ranges due to undefined behavior where two switch cases overlap. ie: make your first range 0 to 100 then your second 101 to 130, etc